sql/item_row.h

```diff
@@ -57,104 +57,104 @@ class Item_row: public Item_fixed_hybrid,
     not_null_tables_cache(0), with_null(0)
   { }
 
-  bool with_subquery() const { DBUG_ASSERT(fixed); return m_with_subquery; }
-  enum Type type() const { return ROW_ITEM; };
-  const Type_handler *type_handler() const { return &type_handler_row; }
+  bool with_subquery() const override { DBUG_ASSERT(fixed); return m_with_subquery; }
+  enum Type type() const override { return ROW_ITEM; };
+  const Type_handler *type_handler() const override { return &type_handler_row; }
   Field *create_tmp_field_ex(MEM_ROOT *root, TABLE *table, Tmp_field_src *src,
-                             const Tmp_field_param *param)
+                             const Tmp_field_param *param) override
   {
     return NULL; // Check with Vicentiu why it's called for Item_row
   }
   void illegal_method_call(const char *);
-  bool is_null() { return null_value; }
-  void make_send_field(THD *thd, Send_field *)
+  bool is_null() override { return null_value; }
+  void make_send_field(THD *thd, Send_field *) override
   {
     illegal_method_call((const char*)"make_send_field");
   };
-  double val_real()
+  double val_real() override
   {
     illegal_method_call((const char*)"val");
     return 0;
   };
-  longlong val_int()
+  longlong val_int() override
   {
     illegal_method_call((const char*)"val_int");
```

Comment:
Side note (to be handled in a separate fix): C-style casts should be avoided, and string literals are already compatible with `const char*` by default.

Some savings could be had if `Item_row::illegal_method_call(const char*)` returned a `size_t` of 0. In that case, tail call optimization could be enabled:
```c++
return illegal_method_call("val_int");
```
or
```c++
return reinterpret_cast<String*>(illegal_method_call("val_str"));
```